.ci.yaml

```
@@ -156,6 +156,17 @@ platform_properties:
       cpu: arm64
       device_os: iOS-16
       xcode: 14c18 # Xcode 14.2 to support iOS 16.2 in devicelab.
+  mac_build_test:
+    properties:
+      dependencies: >-
+        [
+          {"dependency": "xcode", "version": "14c18"},
+          {"dependency": "gems", "version": "v3.3.14"},
+          {"dependency": "apple_signing", "version": "version:2022_to_2023"}
+        ]
+      os: Mac-12
+      device_type: none
+      xcode: 14c18 # Xcode 14.2 to support iOS 16.2 in devicelab.
```

How are the recipes chosen?  If it's devicelab this is fine since @godofredoc has said it has hermetic Xcode caches, and it seems like the `Mac_build_test` tests are going to need tethered devices.  However if it's not the devicelab recipe then it should be

```suggestion
      xcode: 14a5294e # xcode 14.0 beta 5
```

because of the Xcode cache issues: https://github.com/flutter/flutter/issues/117541#issuecomment-1363232756

@godofredoc can you clarify if this is right?

---

Good catch. The parent build is running on a host only bot, and should use same xcode as other host only targets. Updated to use 14a5294e and added comments to clarify

---

The `devicelab_drone_build_test` recipe currently injects the `devicelab_osx_sdk` version based on property `xcode` to run `test`. This will cause the `test` run on a `devicelab` bot but with an `xcode` version defined for the `build` running on the host only bot. This is clearly not expected.

Based on current logics (ci.yaml roller & cocoon scheduler) which auto generate targets' `xcode` (`devicelab_osx_sdk` or `osx_sdk`) based on the platform `xcode`, this PR as of now will not work.

**We expect:**

- `osx_sdk: 14a5294e` for build step on the host only bot
- `devicelab_osx_sdk: 14c18` for test step on the devicelab bot.

**We get:**

- same version for both.

To make this PR work, one way we can do is twist both the `ci.yaml roller` and `cocoon scheduler` to add both `osx_sdk` and `devicelab_osx_sdk` to the target. But this will add additional complexity to existing logics.

On the other hand, I do not see we will have a short-term fix on the xcode issue. To make things easier and unblock here. I am thinking to add `clean cache` flag to host only bots, allowing bots a couple of days to clean up old xcode's cache and prepare for the new version. This way, we will be unblocked here for all devicelab mitgration to build+test, and also make xcode version consistent across host only and devicelab targets. 

With that being said, and considering ongoing frequent xcode changes, we should this to Q2.

@godofredoc before the xcode issue resolved from root (hopefully in Q2), do you have any objection for the above proposal?

---

This is unblocked, and have updated to use the latest xcode: 14e222b.

First run when no artifact exists: https://luci-milo.appspot.com/raw/build/logs.chromium.org/flutter/led/keyonghan_google.com/abc2f7c6ea7e36380cb59c73527839a5f35bd9ea5e51ce7c0d210fe704957246/+/build.proto
Second run when the artifact exists: https://luci-milo.appspot.com/raw/build/logs.chromium.org/flutter/led/keyonghan_google.com/5fe9cb5279aa53d509bb1cce94491cc5911bd9873851302b180e7205de244cb8/+/build.proto

@christopherfujino Please help take a look as @jmagman is ooo. Thanks!